### PR TITLE
Enable UDS settings

### PR DESCRIPTION
**Pull request by r1viollet**
# What does this PR do?

Ensure UDS settings are allowed by the exporter

# Motivation

Allow usage of UDS.
https://github.com/DataDog/ddprof/issues/232

# Testing

Testing is ongoing

### CODE REVIEW

File: include/ddprof_input.hpp [hunk 1]
```diff
@@ -76,9 +75,8 @@ typedef struct DDProfInput {
   XX(DD_API_KEY,                        api_key,            A, 'A', 1, input, NULL, "",           exp_input.)  \
   XX(DD_ENV,                            environment,        E, 'E', 1, input, NULL, "",           exp_input.)  \
   XX(DD_AGENT_HOST,                     host,               H, 'H', 1, input, NULL, "localhost",  exp_input.)  \
-  XX(DD_SITE,                           site,               I, 'I', 1, input, NULL, "",           exp_input.)  \
   XX(DD_TRACE_AGENT_PORT,               port,               P, 'P', 1, input, NULL, "8126",       exp_input.)  \
-  XX(DD_TRACE_AGENT_URL,                url,                U, 'U', 1, input, NULL, "", )                      \
+  XX(DD_TRACE_AGENT_URL,                url,                U, 'U', 1, input, NULL, "",           exp_input.)  \
```
Comment (sanchda) [inline include/ddprof_input.hpp:79]: Ah, makes sense to move this 👍

File: src/exporter/ddprof_exporter.cc [hunk 2]
```diff
@@ -91,20 +115,43 @@ DDRes ddprof_exporter_init(const ExporterInput *exporter_input,
...
-    exporter->_url =
-        alloc_url_agent("http://", exporter_input->host, exporter_input->port);
+    const char *port_str = exporter_input->port;
+
+    if (exporter_input->url) {
+      // uds -> no port
+      if (!strncasecmp(exporter_input->url, "unix", 4)) {
+        port_str = nullptr;
+      }
+      // already port -> no port
```
Comment (sanchda) [inline src/exporter/ddprof_exporter.cc:125]: yup, agreed

File: src/exporter/ddprof_exporter.cc [hunk 3]
```diff
@@ -91,20 +115,43 @@ DDRes ddprof_exporter_init(const ExporterInput *exporter_input,
   }
 
   if (exporter->_agent) {
-    exporter->_url =
-        alloc_url_agent("http://", exporter_input->host, exporter_input->port);
+    const char *port_str = exporter_input->port;
```
Comment (nsavoire) [inline src/exporter/ddprof_exporter.cc:118]: This should be probably be extracted to its own function.
I would simplify the logic by sopping mixing url and port:
* either a url is given in input, in that case it's expected to be a full url (with protocol and port), in that host and port are ignored
* either no url is given in input and url is built from 'http', host and port

That way you can even stop distinguishing between agent and agentless.

**Review by sanchda — APPROVED** [review]
I think this looks good.  At some point we should probably change the input stuff to using std::string or something instead of raw C strings, but I don't think this is really the time or the place for that.  :)

Good stuff!  Thank you!  UDS support may not improve performance substantially, since we use async, but it's still a configuration users will want!

**Review by nsavoire — APPROVED** [review]

File: src/exporter/ddprof_exporter.cc (hunk 3 above)
Comment (r1viollet) [reply src/exporter/ddprof_exporter.cc:118]: This is a good idea, though we would need to check if endpoint is reachable on startup,
Adding this permissive config might be bad for reverting it later.

**Comment by r1viollet** [comment]
Merging as is for now, I agree there should be a follow up on this.
The way to configure this is not really explicit as of now. The documentation needs some love. FYI:

```
unix:///run/apm.sock
```

**Pull request merged**